pkg/network/server.go

```diff
@@ -404,6 +428,20 @@ func (s *Server) handleBlockCmd(p Peer, block *core.Block) error {
 	return s.bQueue.putBlock(block)
 }
 
+// handlePing processes ping request.
+func (s *Server) handlePing(p Peer, ping *payload.Ping) error {
+	return p.WriteMsg(NewMessage(s.Net, CMDPong, payload.NewPing(s.id, s.chain.BlockHeight())))
+}
+
+// handlePing processes pong request.
+func (s *Server) handlePong(p Peer, pong *payload.Ping) error {
+	s.pingSent = false
+	if s.chain.HeaderHeight() < pong.LastBlockIndex {
```

Comment:
I think something should update peer's `lastBlockIndex` here.

Reply:
reworked this part a bit. Updating last block index and checking whether we had to receive pong =)